### PR TITLE
feat: Read toolsId from Adaptation Projects ui5.yaml file in telemetry package

### PR DESCRIPTION
**Pull request by nikmace**
Feat for #1847.

- Enhances source template retrieval for Adaptation Projects. If the ui5.yaml file does not have a support section default values will be returned (i.e NO_TOOLS_ID).

### CODE REVIEW

**Comment by nikmace** [comment]
I would suggest waiting until this one gets approved and merged: https://github.com/SAP/open-ux-tools/pull/1794. But this PR does not break anything to be fair.

File: packages/telemetry/src/tooling-telemetry/data-processor.ts
Comment (devinea) [inline packages/telemetry/src/tooling-telemetry/data-processor.ts:276]: @nikmace Why support both `manifest.appdescr_variant` and `ui5.yaml` for adaptation projects?

Comment (nikmace) [reply packages/telemetry/src/tooling-telemetry/data-processor.ts:276]: Here we need to determine what project type is being run. Adaptation Projects have `manifest.appdescr_variant` file and fiori apps have `manifest.json`. For Adaptation Projects we decided to put the toolsId inside `ui5.yaml` file. Or I need to prop drill and pass the project type this method.

Comment (devinea) [reply packages/telemetry/src/tooling-telemetry/data-processor.ts:276]: Thanks @nikmace for explanation.

**Review by devinea — DISMISSED** [review]
Code changes to support toolsid in ui5.yaml for adaptation projects looks good. 
Covered by tests
Changeset 👍

File: packages/telemetry/src/tooling-telemetry/data-processor.ts
Comment (GDamyanov) [inline packages/telemetry/src/tooling-telemetry/data-processor.ts:286]: I think that here will be good also to include `&& fs.existsSync(paths.ui5Yaml)` in the condition of the if statement

File: packages/telemetry/src/tooling-telemetry/data-processor.ts
Comment (GDamyanov) [inline packages/telemetry/src/tooling-telemetry/data-processor.ts:293]: Would it be better to log the error, instead of have empty catch statement

**Review by GDamyanov — CHANGES_REQUESTED** [review]

File: packages/telemetry/src/tooling-telemetry/data-processor.ts
Comment (nikmace) [reply packages/telemetry/src/tooling-telemetry/data-processor.ts:293]: I think it was like this before too because we don't really care if something failed there. In the end, there are default properties that are returned. Also, there is no logging in to this package.

**Review by GDamyanov — APPROVED** [review]
- Comments addressed
- Simple and easy to understand code
- Good test coverage

**Comment by nikmace** [comment]
Hello, @devinea. Would you mind re-reviewing or re-approving this one so it could get merged?

**Review by devinea — APPROVED** [review]
Re-approving. Thanks

**Pull request merged**
